src/main/java/seedu/mtracker/model/Instrument.java

```diff
@@ -1,30 +1,42 @@
 package seedu.mtracker.model;
 
+import java.util.HashMap;
+import java.util.HashSet;
+
 public abstract class Instrument {
     protected String name;
     protected double currentPrice;
     protected String sentiment;
     protected boolean isDone;
 
-    protected static final String EMPTY_STRING = "";
-    protected static final String FILE_SEPARATOR = ";";
+    protected static HashSet<String> validAttribute;
+
     public static final String SEMICOLON_SEP = "; ";
     public static final String SPACE = " ";
+    protected static final String EMPTY_STRING = "";
+    protected static final String FILE_SEPARATOR = ";";
 
-    private static final String TYPE_FIELD = "Type: ";
-    private static final String NAME_FIELD = "Name: ";
-    private static final String CURRENT_PRICE_FIELD = "Current Price: ";
-    private static final String SENTIMENT_FIELD = "Sentiment: ";
+    protected static final String TYPE_FIELD = "Type: ";
+    protected static final String NAME_FIELD = "Name: ";
+    protected static final String CURRENT_PRICE_FIELD = "Current Price: ";
+    protected static final String SENTIMENT_FIELD = "Sentiment: ";
     protected static final String REMARKS_FIELD = "Remarks: ";
 
     protected static final String DONE_SYMBOL = "[X]";
     protected static final String NOT_DONE_SYMBOL = "[ ]";
 
+    protected static final String NAME_ATTRIBUTE = "name";
+    protected static final String CURRENT_PRICE_ATTRIBUTE = "current-price";
+    protected static final String SENTIMENT_ATTRIBUTE = "sentiment";
+    protected static final String REMARK_ATTRIBUTE = "remark";
+    protected static final String SEPARATOR = " , ";

```

Comment:
```suggestion
    protected static final String SEPARATOR = ", ";
```
is there a reason for space?